pkg/agent/agent.go

```diff
@@ -845,6 +848,15 @@ func (i *Initializer) allocateGatewayAddresses(localSubnets []*net.IPNet, gatewa
 	if err := util.ConfigureLinkAddresses(i.nodeConfig.GatewayConfig.LinkIndex, gwIPs); err != nil {
 		return err
 	}
+	// Periodically check whether IP configuration of the gateway is correct.
+	// Terminated when stopCh is closed.
+	if !i.networkConfig.TrafficEncapMode.IsNetworkPolicyOnly() {
```

Comment:
When it gets here, this check has been done. And since L848 already configures link addresses, logically it's ok to just start a goroutine to do the periodical work without any check.